flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPoolTest.java

```diff
@@ -515,6 +518,27 @@ public void testIsAvailableOrNot() throws InterruptedException {
         assertTrue(availableFuture2.isDone());
     }
 
+    /** For FLINK-20547: https://issues.apache.org/jira/browse/FLINK-20547. */
+    @Test
+    public void testConsistentAvailability() throws Exception {
+        MemorySegment segment = null;
+        BufferPool localPool = null;
+        NetworkBufferPool globalPool = new TestNetworkBufferPool(numBuffers, memorySegmentSize);
+        try {
+            localPool = new LocalBufferPool(globalPool, 1);
+            segment = localPool.requestBufferBuilderBlocking().getMemorySegment();
+            localPool.setNumBuffers(2);
+        } finally {
+            if (segment != null) {
+                localPool.recycle(segment);
+            }
+            if (localPool != null) {
+                localPool.lazyDestroy();
+            }
+            globalPool.destroy();
+        }
```

Comment:
nit: I think it would be slightly simpler to use `CloseableRegistry`?
```
try (closeableRegistry = new CloseableRegistry()) {
  ...
}
```

Reply:
I simplify the code without using CloseableRegistry.